### PR TITLE
DAOS-9314 container: only stop started container.

### PR DESCRIPTION
**Pull request by wangdi1**
Some ds_cont_child will only created by ds_cont_child_lookup(),
never be started at all, so let's check if the container child
is started before stop the container child.

Signed-off-by: Di Wang <di.wang@intel.com>

### CODE REVIEW

File: src/container/srv_target.c [hunk 1]
```diff
@@ -843,7 +843,10 @@ cont_child_started(struct ds_cont_child *cont_child)
 static void
 cont_child_stop(struct ds_cont_child *cont_child)
 {
-	if (!cont_child->sc_stopping) {
+	/* Some ds_cont_child will only created by ds_cont_child_lookup().
+	 * never be started at all
+	 */
+	if (!cont_child->sc_stopping && cont_child_started(cont_child)) {
```
Comment (NiuYawei) [inline src/container/srv_target.c:849]: Could you explain in detail why "Some ds_cont_child will only created by ds_cont_child_lookup()"? The stacktrace in DAOS-9314 is in a enumeration handler, which means the container must be opened, I don't see how can it not being started.

Maybe we'd add a D_ASSERT(cont_child_started(cont_child)) instead?

**Review by NiuYawei — CHANGES_REQUESTED** [review]

**Comment by wangdi1** [comment]
>Could you explain in detail why "Some ds_cont_child will only created by ds_cont_child_lookup()"? 

I assume  the container is destroyed, but got DER_BUSY, though cont_child_stop() is being called, so it is removed from the LRU cache, 

In the mean time (or even after), if there is migrate fetch/enumerate rpc arrive, then in  obj_ioc_init(), it will call ds_cont_child_look(), which will allocate the container child by ds_cont_child_lookup() without starting the container.  

Then if the user destroy the container again, it will hit this issue. I think

It looks like so according to the job.log & daos server log.

**Review by NiuYawei — DISMISSED** [review]

**Review by liw — APPROVED** [review]

File: src/container/srv_target.c [hunk 2]
```diff
@@ -855,8 +858,6 @@ cont_child_stop(struct ds_cont_child *cont_child)
 		/* cont_stop_agg() may yield */
 		cont_stop_agg(cont_child);
 		ds_cont_child_put(cont_child);
-	} else {
-		D_ASSERT(!cont_child_started(cont_child));
```
Comment (gnailzenh) [inline src/container/srv_target.c:859]: but this is not the assertion we got right?

Comment (wangdi1) [reply src/container/srv_target.c:859]: oh, not this assertion, but this assertion become useless with the patch, so I removed it.

**Comment by wangdi1** [comment]
Not sure why it show failure on Hardware medium. I do not see any failure there.

**Comment by wangdi1** [comment]
ping

**Review by NiuYawei — APPROVED** [review]

**Review by liw — APPROVED** [review]

**Pull request merged**